lib/internal/streams/readable.js

```diff
@@ -1287,6 +1361,234 @@ ObjectDefineProperties(ReadableState.prototype, {
   }
 });
 
+function isLocked(self) {
+  return self[kConsume]?.stream?.locked === true;
```

Comment:
I would avoid these `?`. On some other benchmarks they were not great perf-wise.